main.py

```diff
@@ -15,28 +16,89 @@ class Robot:
     UPPER_POSITION = 30
     LOWER_POSITION = 0
     WIGGLE_AMOUNT = 5
+    INACTIVITY_PERIOD_S = 15
 
     async def connect(self):
         opts = RobotClient.Options(
             refresh_interval=0,
             dial_options=DialOptions(credentials=secrets.creds)
         )
         self.robot = await RobotClient.at_address(secrets.address, opts)
-        self.servo = Servo.from_robot(self.robot, "servo")
+        self._servo = Servo.from_robot(self.robot, "servo")
+
+        self._mutex = threading.Lock()
+        self._count = 0  # Number of hands currently raised
+
+        # We have a background thread that wiggles the hand when it's been
+        # raised for a long time. This condition variable is how to shut that
+        # down.
+        # It would be intuitive to pass self._mutex to threading.Condition, but
+        # that will result in deadlock when you have the mutex and wait for the
+        # thread to shut down.
+        self._cv = threading.Condition()
+        self._should_shutdown_thread = True
+        self._thread = None
+
+    def _start_thread(self):
+        self._should_shutdown_thread = False
+        if self._thread is not None:
+            print("LOGIC BUG: starting the thread when it's already started!?")
+        self._thread = threading.Thread(
+                target=asyncio.run,
+                args=(self._wiggle_on_inactivity(),),
+                daemon=True)
+        self._thread.start()
+
+    def _stop_thread(self):
+        self._should_shutdown_thread = True
+        with self._cv:
+            self._cv.notify()
+        self._thread.join()
+        self._thread = None
 
     async def raise_hand(self):
-        await self.servo.move(self.UPPER_POSITION)
+        with self._mutex:
+            self._count += 1
+            if self._count == 1:
+                await self._servo.move(self.UPPER_POSITION)
+                self._start_thread()
 
     async def lower_hand(self):
```

Comment:
Perhaps these functions need docstrings now. Their meaning has changed from the naive "raise/lower the robot's hand" to "consider someone in the audience to have raised/lowered their hand."